src/main/java/ch/njol/skript/sections/recipe/SecCookingRecipe.java

```diff
@@ -0,0 +1,192 @@
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.sections.recipe;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.config.EntryNode;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.config.validate.SectionValidator;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.log.RetainingLogHandler;
+import ch.njol.skript.log.SkriptLogger;
+import ch.njol.skript.patterns.PatternCompiler;
+import ch.njol.skript.patterns.SkriptPattern;
+import ch.njol.skript.util.Timespan;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import com.google.common.collect.Iterables;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.BlastingRecipe;
+import org.bukkit.inventory.CampfireRecipe;
+import org.bukkit.inventory.CookingRecipe;
+import org.bukkit.inventory.FurnaceRecipe;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.RecipeChoice;
+import org.bukkit.inventory.ShapedRecipe;
+import org.bukkit.inventory.ShapelessRecipe;
+import org.bukkit.inventory.SmokingRecipe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+@Name("Cooking Recipe")
+@Description("Creates a cooking recipe")
+@Examples({
+	"create a crafting recipe for diamond named \"Dirty Diamond\" with the key \"dirty-diamond\":",
+	"\tshape:",
+	"\t\tdirt, dirt and dirt",
+	"\t\tdirt, diamond and dirt",
+	"\t\tdirt, dirt and dirt"
+})
+@Since("INSERT VERSION")
+public class SecCookingRecipe extends Section {
+
+	static {
+			Skript.registerSection(SecCookingRecipe.class, "(create|add|register) [a] (0:blast furnace|1:campfire|2:furnace|3:smoker) recipe for %itemtype% with [the] key %string%");
+	}
+
+	private static SectionValidator validator = new SectionValidator()
+		.addEntry("ingredient", false)
+		.addEntry("cook time", false)
+		.addEntry("group", true)
+		.addEntry("xp", false);
+
+
+	enum CookingRecipeType {
+		BLAST_FURNACE, CAMPFIRE, FURNACE, SMOKER
+	}
+
+	private Expression<String> key;
+	private CookingRecipeType type;
+	private Expression<? extends ItemType> ingredient;
+	private Expression<? extends Timespan> cookTime;
+	private Expression<? extends Number> xp;
+	private Expression<ItemType> result;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult, SectionNode sectionNode, List<TriggerItem> triggerItems) {
```

Comment:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, SectionNode sectionNode, List<TriggerItem> triggerItems) {
```
import update required